text/0000-entity-literal-syntax.md

```diff
@@ -0,0 +1,224 @@
+
+### Option 2: instance expressions
+
+The main alternative would be an inline syntax. As a separate syntax, it could look like this:
+
+```
+[
+  Namespace::SomeType::"entity_id"([Namespace::OtherType::"parent_entity_id"], {
+      attribute: "value"
+    }, {
+      tagName: "value"
+    }),
+  //subsequent entities
+]
+```
+
+This would be more explicit that the content of the file is a set of entities and only a set of entities (like with JSON). It's a little less familiar/similar to schema syntax than Option 1, and might be harder to do useful syntax highlighting, but it is relatively close to how extension types like `datetime` are written. It opens up the possibility of composition within policies and schemas, but that is only relevant if that would ever be needed. This syntax doesn't allow for reusable records.
+
+### Option 3: instance set declaration
+
+A third alternative that would allow for inline literals _and_ reusable records would be to define a keyword for the set, say `entities`:
+
+```
+entities [
+  Namespace::SomeType::"entity_id"([Namespace::OtherType::"parent_entity_id"], {
+      attribute: "value",
+      recordAttr: myRecord
+    }, {
+      tagName: "value"
+    }),
+  //subsequent entities
+]
+
+record myRecord = {
+  foo: "bar"
+};
+```
+
+This could allow for namespace blocks, but a complication is whether there can be multiple `entities` declarations in a file or only one.
+
+## Drawbacks
+
+There are several reasons not to develop a custom syntax:
+
+### Multiple formats can raise cognitive burden
+
+Adding another custom format raises the bar for what customers need to know. They may ignore one format at first, but eventually they may need to learn both. For example, if one team uses the JSON format and another uses the custom format, but then the teams merge, they will each end up having to read/update the other's format. They may also fight about which format to move to.
+
+Mitigating this problem is that it's easy and predictable to convert between the two formats, since the syntactic concepts line up very closely. The features you lose when converting from the new format to the JSON one would be 1/ any comments, and 2/ any use of intermingling of `action`, `entity`, and `type` declarations, since they must all be in their own sections in the JSON. Otherwise the features line up very closely and the only difference is syntax. We would expect to write conversion tools as part of implementing the new syntax (which are easily achieved by parsing in the new format and pretty-printing in JSON).
```

Comment:
I agree that easy conversion between the formats is a sufficient mitigation for this. And that everything said in the Motivation section outweighs the drawback of increased cognitive burden described here.